experimental/GModule/src/GModule.jl

```diff
@@ -832,6 +833,74 @@ function _irred_abelian(G::Oscar.GAPGroup)
   return all
 end
 
+# representatives of `F`-irreducible representations of a finite abelian group
+# `G` over the field `F` of size `p^k`
+function _irred_abelian(G::Oscar.GAPGroup, F::FinField)
+  @assert is_abelian(G)
+  q = order(F)
+  p = characteristic(F)
+  k, _ = remove(q, p)
+
+  if mod(order(G), p) != 0
+    GG = G
+  else
+    # Replace `G` by its Hall p'-subgroup,
+    # such that the generators correspond to those of `G`.
+    # (In the end, we will construct modules,
+    # there is no need to create an epimorphism to the largest p'-factor.)
+    function pprime_part(g, p)
+      v, _ = remove(order(g), p)
```

Comment:
Do you know `ppio`? `ppio(a, b) = gcd(a, b^infty)`, so here `ppio(q, p) = p^k` for `k = valuation(q, p)`

Reply:
Thanks.
`ppio` is a very suggestive name.
(Searching for it in the Oscar manual yields hundreds of matches, most (all?) of them have nothing to do with `ppio`.)